src/check_decl.cpp

```diff
@@ -180,6 +180,8 @@ gb_internal void override_entity_in_scope(Entity *original_entity, Entity *new_e
 
 	original_entity->flags |= EntityFlag_Overridden;
 	original_entity->type = new_entity->type;
+	original_entity->kind = new_entity->kind;
+	original_entity->decl_info = new_entity->decl_info;
```

Comment:
I'm not sure if this is entirely correct, assuming the comment above holds...
I've found that changing the `kind` is a must otherwise a chain of aliases result in e.g.: `Constant -> Constant -> Procedure` entity types.

The change on `decl_info` is due to other parts of the code asserting that an entity of `kind` holds matching `decl_info` which does not hold with aliases.